docs/modules/ROOT/pages/components.adoc

```diff
@@ -0,0 +1,186 @@
+= Using Components
+
+The following documentation provides reasoning and examples on how to use Contracts for Cairo components.
+
+== Components
+
+:shamans-post: https://community.starknet.io/t/cairo-components/101136#components-1[Starknet Shamans post]
+
+Starknet components are separate modules that contain storage, events, and implementations that can be integrated into a contract.
+Components themselves cannot be declared or deployed.
+Another way to think of components is that they are abstract modules that must be instantiated.
+
+TIP: For more information on the construction and design of Starknet components, see the {shamans-post}.
+
+=== Setting up
+
+:initializable-component: xref:/security.adoc#initializable[InitializableComponent]
+
+Integrating a component first requires that the contract declares the component with the `component!` macro:
+
+[,javascript]
+----
+#[starknet::contract]
+mod MyContract {
+    // Import the Contracts for Cairo component
```

Comment:
```suggestion
    // Import the component
```

I don't like the phrase "the Contracts for Cairo component". it seems unnecessarily large imo and is used in other places as well. We could replace it with something like the library component, or a similar shorter version maybe.